client/state/domains/management/reducer.js

```diff
@@ -72,22 +78,18 @@ export const saving = createReducer( false, {
  * @return {Object}        Updated state
  */
 export const items = createReducer( {}, {
-	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_RECEIVE ]: ( state, { cacheData } ) => ( { ...state, _contactDetailsCache: cacheData } ),
-	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_UPDATE ]:
-		( state, { cacheData } ) =>
-			merge( {}, state, { _contactDetailsCache: cacheData } ),
+	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_RECEIVE ]: ( state, { data } ) => ( { ...state, _contactDetailsCache: data } ),
+	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_UPDATE ]: ( state, { data } ) => {
+		return merge( {}, state, { _contactDetailsCache: data } );
```

Comment:
I'll look at this again sometime later as I'm firefighting something else atm, but offhand I don't think that works due to the payload in `data`. There is nesting in the incoming structure and IIRC (and totally may be wrong here) spread solutions did not work.

Perhaps that's symptomatic of something less than idea we're doing in constructing the data structure as I think you alluded to below. I'll have to circle back.